hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataWriter.java

```diff
@@ -60,4 +62,12 @@
    */
   void update(HoodieRollbackMetadata rollbackMetadata, String instantTime);
 
+  /**
+   * Drop the given metadata partitions. This path reuses DELETE_PARTITION operation.
+   *
+   * @param instantTime - instant time when replacecommit corresponding to the drop will be recorded in the metadata timeline
+   * @param partitions - list of {@link MetadataPartitionType} to drop
+   * @throws IOException
+   */
+  void dropPartitions(String instantTime, List<MetadataPartitionType> partitions);
```

Comment:
can we maintain the same name as writeClient. deletePartitions instead of dropPartitions.